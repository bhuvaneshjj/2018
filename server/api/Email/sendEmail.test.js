'use strict';

const { API_SEND_EMAIL, apiCall } = require('../../../common/common');
const createWsServer = require('../../createWsServer');
const createWsClient = require('./../createWsClient');
const generateTestToken = require('../generateTestToken');

const port = 5613;
const wsServer = createWsServer({});
const wsClient = createWsClient({ port });
const token = generateTestToken();

beforeAll(async () => {
  wsServer.listen(port);
  await wsClient.open();
});

afterAll(async () => {
  await wsClient.close();
  await wsServer.close();
});

it('sendEmail successfully', async () => {
  const { requestId, ...response } = await wsClient.sendRequest(
    apiCall({
      endpoint: API_SEND_EMAIL,
      request: {
        mailTo: 'test@test.io',
        subject: 'Test subject',
        html: 'Test email for <i>masses</i>.'
      },
      token
    })
  );

  expect(response).toMatchSnapshot();
});

it('not authenticated', async () => {
  const { requestId, ...response } = await wsClient.sendRequest(
    apiCall({ endpoint: API_SEND_EMAIL })
  );
  expect(response).toMatchSnapshot();
});
