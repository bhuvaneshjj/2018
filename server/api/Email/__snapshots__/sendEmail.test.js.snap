// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not authenticated 1`] = `
Object {
  "code": "authentication token invalid",
  "status": "Špatný ověřovací token. Zkus se přihlásit znovu. Detaily: JsonWebTokenError: jwt must be provided",
}
`;

exports[`sendEmail successfully 1`] = `
Object {
  "code": "ok",
}
`;
